Admin/CrossTenantMailboxMigrationValidation.ps1

```diff
@@ -1001,15 +1045,17 @@ if ($CheckObjects -and !$NoConn -and !$SourceIsOffline) {
 if ($CheckOrgs -and !$SourceIsOffline) {
     LoggingOn
     ConnectToSourceTenantAAD
-    $SourceTenantId = (Get-AzureADTenantDetail).ObjectId
-    Write-Verbose -Message "Informational: SourceTenantId gathered from AzureADTenantDetail.ObjectId: $SourceTenantId"
-    $SourceAADApp = Get-AzureADServicePrincipal -All $true
-    Write-Verbose -Message "Informational: AAD apps gathered from AzureADServicePrincipal"
+    $SourceTenantId = (Get-MgOrganization).id
+    Write-Verbose -Message "Informational: SourceTenantId gathered from (Get-MgOrganization).id: $SourceTenantId"
+    $SourceAADApp = Get-MgServicePrincipal -All
+    Write-Verbose -Message "Informational: AAD apps gathered from MgServicePrincipal"
+    Disconnect-MgGraph | Out-Null
     ConnectToTargetTenantAAD
-    $TargetTenantId = (Get-AzureADTenantDetail).ObjectId
-    Write-Verbose -Message "Informational: TargetTenantId gathered from AzureADTenantDetail.ObjectId: $TargetTenantId"
+    $TargetTenantId = (Get-MgOrganization).id
+    Write-Verbose -Message "Informational: TargetTenantId gathered from (Get-MgOrganization).id: $TargetTenantId"
     Write-Verbose -Message "Informational: Checking if there's already an AAD Application on TARGET tenant that meets the criteria"
-    $TargetAADApp = Get-AzureADApplication | Where-Object { ($_.ReplyUrls -eq "https://office.com") -and ($_.RequiredResourceAccess -like "*ResourceAppId: 00000002-0000-0ff1-ce00-000000000000*") }
+    $TargetAADApp = Get-MgApplication | Where-Object { ($_.Web.RedirectUris -eq "https://office.com") -and ($_.RequiredResourceAccess.ResourceAppId -like "*00000002-0000-0ff1-ce00-000000000000*") }
+    Disconnect-MgGraph | Out-Null
```

Comment:
Why are we doing a couple of `Disconnect-MgGraph` here when it looks like we us it again right afterwords?

Reply:
yup, that's because I connect to AAD against the source tenant first, and then against the target tenant, with their proper disconnect to kill the MgGraph sessions cleanly. (lines 1047 and 1067)